tools/wake-format/emitter.cpp

```diff
@@ -1251,7 +1257,7 @@ wcl::doc Emitter::walk_paren(ctx_t ctx, CSTElement node) {
                    .token(TOKEN_P_PCLOSE)
                    .format(ctx, node.firstChildElement(), token_traits);
 
-  if (!no_nl->has_newline()) {
+  if (no_nl->newline_count() == count_allowed_newlines(token_traits, node)) {
```

Comment:
Is there ever going to be a case where `no_nl` will have *dropped* some of the allowed newlines?

Reply:
nope! The "allowed newlines" are essentially the newlines emitted from comments immediately before and after the subtree being emitted. Those comments are automatically emitted via the API and can't be skipped.